pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/inv/RangeIndexCreator.java

```diff
@@ -165,26 +166,63 @@ public RangeIndexCreator(File indexDir, FieldSpec fieldSpec, FieldSpec.DataType
   }
 
   @Override
-  public void add(int dictId) {
-    _numberValueBuffer.put(_nextDocId, dictId);
-    _docIdBuffer.put(_nextDocId, _nextDocId);
-    _nextDocId = _nextDocId + 1;
+  public void add(int value) {
+    addValueToBuffer(value);
+    nextDoc();
   }
 
   @Override
-  public void add(int[] dictIds, int length) {
+  public void add(int[] values, int length) {
     for (int i = 0; i < length; i++) {
-      int dictId = dictIds[i];
-      _numberValueBuffer.put(_nextValueId, dictId);
-      _docIdBuffer.put(_nextValueId, _nextDocId);
-      _nextValueId = _nextValueId + 1;
+      addValueToBuffer(values[i]);
+      nextDoc();
```

Comment:
(CRITICAL) This is not correct. Here you should add to `_nextValueId` instead of `_nextDocId`

Reply:
done